### PR TITLE
Docs: Add API style/casing rule

### LINKED ISSUE #19620: Docs: Add style rule for correct casing of API names
**What would you like to be added**:

I suggest the we treat APIs as products, and that when referring to an API, it should be written in Title Case, and followed by "API"—for example, "Dashboard Permissions API".

This should be added to the [docs style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md).

### PR DESCRIPTION
**Pull request by HG00**
**What this PR does / why we need it**: Adds a rule to style guide for API names

**Which issue(s) this PR fixes**: 

Fixes #19620

**Special notes for your reviewer**: Taking a stab at this, seemed to be an appropriate heading for it, and didnt want to make a new heading for just one line

### CODE REVIEW

**Comment by torkelo** [comment]
please put a real title on this PR

**Comment by HG00** [comment]
> please put a real title on this PR

Hows that?

**Comment by marefr** [comment]
Updated list of `Fixes` based on references in commits so that those issues will be closed when this PR is merged. 

Feedback. It's normally desirable with one PR targeting to fix one issue as it makes it easier to review the changes.

**Comment by marefr** [comment]
Oops. My bad. Just realized that there was only one commit :+1: But got confused of the list commits showing in the activity of this PR. I removed the issue number from the PR title as it's not according to our naming rules, see https://github.com/grafana/grafana/blob/master/contribute/pull-request.md#pull-request-titles.

Looks good now

**Comment by marefr** [comment]
@HG00 one question that you may optionally answer. Are you participating in [Hacktoberfest](https://hacktoberfest.digitalocean.com/) and did you find this issue by looking at the `Hacktoberfest` label?

**Comment by HG00** [comment]
> 
> 
> @HG00 one question that you may optionally answer. Are you participating in [Hacktoberfest](https://hacktoberfest.digitalocean.com/) and did you find this issue by looking at the `Hacktoberfest` label?

Hey @marefr - "sortof", i found a bug in the doc for Centrify, thought "I should submit that" then forgot - then got reminded of hacktoberfest and figured it was a good time to send some in :) - I am participating, does that change anything?

**Comment by marefr** [comment]
@HG00 no, that's definitely not changing anything. Just trying to understand how Grafana's contributors  find issues to work on and/or if Hacktoberfest label actually helps with that.

**Comment by HG00** [comment]
> 
> 
> @HG00 no, that's definitely not changing anything. Just trying to understand how Grafana's contributors find issues to work on and/or if Hacktoberfest label actually helps with that.

For me it did, TBH I think i would have forgotten otherwise. I've also encouraged a few other people to see where they can help, and that #Hacktoberfest helped (along with #help-wanted)

File: contribute/style-guides/documentation-style-guide.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ For all items not covered in this guide, refer to the [Microsoft Style Guide](ht
 ### Capitalization
 
 * Grafana, Loki, and Prometheus are always capitalized unless part of a code block.
+* API names are always Title Case, and followed by "API" — for example, "Dashboard Permissions API"
```
Comment (marcusolsson) [inline contribute/style-guides/documentation-style-guide.md]:
```suggestion
* API names are always Title Case, followed by "API"—for example, "Dashboard Permissions API"
```

**Review by marcusolsson — CHANGES_REQUESTED** [review]
Thanks!

File: contribute/style-guides/documentation-style-guide.md (hunk 1 above)
Comment (HG00) [reply contribute/style-guides/documentation-style-guide.md]: I think thats committed now, anything else i need to do?

**Review by marcusolsson — APPROVED** [review]

**Pull request merged**